core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1278,11 +1287,22 @@ private[spark] object SparkSubmitUtils {
       remoteRepos: Option[String],
       ivyPath: Option[String]): IvySettings = {
     val file = new File(settingsFile)
-    require(file.exists(), s"Ivy settings file $file does not exist")
-    require(file.isFile(), s"Ivy settings file $file is not a normal file")
+    // When running driver in cluster mode, the settingsFile is localized and so needs to be
+    // accessed using just the file name and not the full file path
+    val localizedFile = new File(file.getName)
+    val resolvedFile = if (file.exists()) {
```

Comment:
I think you should rather have a configuration to point out Ivy settings, and set together with `spark.files`.

Reply:
+1 that if we go the `spark.files` route we should probably keep a separate (internal) config to pass the file name info.  Similar to `spark.yarn.user.jar`, `spark.yarn.cache.filenames`, etc.

Reply:
I updated the approach taken by this PR. The implementation is much cleaner IMO, albeit it only handles YARN for now. I am not sure if other resource managers would require a similar feature.